src/EditorFeatures/Core/ExternalAccess/EditorConfig/Features/Completion/CompletionHandler.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings;
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Data;
+using Roslyn.Utilities;
+using System.Linq;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.EditorConfigSettings.Data;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Options;
+using System.Net.WebSockets;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.EditorConfig.Features
+{
+    [ExportStatelessLspService(typeof(CompletionHandler), ProtocolConstants.EditorConfigLanguageContract), Shared]
+    [Method(Methods.TextDocumentCompletionName)]
+    internal class CompletionHandler : IRequestHandler<CompletionParams, CompletionList?>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CompletionHandler()
+        {
+        }
+
+        public bool MutatesSolutionState => false;
+        public bool RequiresLSPSolution => true;
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(CompletionParams request) => request.TextDocument;
+
+        public async Task<CompletionList?> HandleRequestAsync(CompletionParams request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.AdditionalDocument;
+            Contract.ThrowIfNull(document);
+
+            var workspace = context.Solution?.Workspace;
+            Contract.ThrowIfNull(workspace);
```

Comment:
can just do document.Project.Solution.Workspace, which will give you a non-nullable reference